### PR TITLE
 Add `CultureReplacerMiddleware` for use in functional test web sites

### PR DESCRIPTION
**Pull request by dougbu**
- #1766
- nit: Clean up duplicate tests
- see individual commits for additional detail

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs [hunk 1]
```diff
@@ -103,9 +103,8 @@ private static Stream GetMemoryStream(string content)
         }
 
         [Theory]
-        [InlineData(10000)]
+        [InlineData(10000)] // expected failure: same time and length (PranavK)
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs]: @pranavkm what does this comment mean?  is it still relevant?

File: samples/TagHelperSample.Web/project.json [hunk 2]
```diff
@@ -10,7 +10,6 @@
         "Kestrel": "1.0.0-*",
         "Microsoft.AspNet.Mvc": "6.0.0-*",
         "Microsoft.AspNet.Mvc.TagHelpers": "6.0.0-*",
-        "Microsoft.AspNet.Mvc.TestConfiguration": "1.0.0",
```
Comment (rynowak) [inline samples/TagHelperSample.Web/project.json]: why this change? this seems necessary

File: test/WebSites/Microsoft.AspNet.Mvc.TestConfiguration/project.json [hunk 3]
```diff
@@ -1,6 +1,8 @@
 {
     "dependencies": {
         "Microsoft.AspNet.Http": "1.0.0-*",
+        "Microsoft.AspNet.Http.Extensions": "1.0.0-rc1-*",
```
Comment (rynowak) [inline test/WebSites/Microsoft.AspNet.Mvc.TestConfiguration/project.json]: should be `1.0.0-*`

**Comment by rynowak** [comment]
:shipit: from me, still want to see the answer about the duplicate test from @pranavkm

File: samples/TagHelperSample.Web/project.json (hunk 2 above)
Comment (dougbu) [inline samples/TagHelperSample.Web/project.json]: just a nit; the dependency isn't needed so I removed it from the two samples

**Comment by dougbu** [comment]
Rebased and correct dependency version

File: test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs]: It comes from here - https://github.com/aspnet/Mvc/commit/6600e68fc0d64e382f9172bfc4da1cbcb12bee10#diff-64a550e80195e44d72c7cba6ea99f428R107. I think you can drop it - the test name should suffice in explaining what scenario this is.

Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs]: Also, why change this - it seems completely unrelated to your PR.

File: samples/TagHelperSample.Web/project.json (hunk 2 above)
Comment (pranavkm) [inline samples/TagHelperSample.Web/project.json]: Let's make this change as part of a separate PR in that case.

File: samples/MvcSample.Web/project.json [hunk 4]
```diff
@@ -9,7 +9,6 @@
     "dependencies": {
         "Kestrel": "1.0.0-*",
         "Microsoft.AspNet.Mvc": "6.0.0-*",
-        "Microsoft.AspNet.Mvc.TestConfiguration": "1.0.0",
```
Comment (pranavkm) [inline samples/MvcSample.Web/project.json]: Undo.

File: test/Microsoft.AspNet.Mvc.Core.Test/OptionDescriptors/DefaultValidationExcludeFiltersProviderTests.cs [hunk 5]
```diff
@@ -39,7 +39,6 @@ public void Add_WithType_RegistersTypesAndDerivedType_ToBeExcluded(Type type)
         [Theory]
         [InlineData(typeof(BaseType))]
         [InlineData(typeof(UnRelatedType))]
-        [InlineData(typeof(UnRelatedType))]
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Core.Test/OptionDescriptors/DefaultValidationExcludeFiltersProviderTests.cs]: Undo.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs [hunk 6]
```diff
@@ -592,16 +592,6 @@ public static IEnumerable<object[]> QueryParameters
                         ""
                     };
 
-                // Attribute Route, manId:int:min(10)? constraint
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs]: Why did this change?

**Comment by pranavkm** [comment]
Looks good apart from the unrelated changes. :shipit: once those are addressed.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs (hunk 6 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs]: :+1: Duplicate test

File: test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs]: :information_source: the duplicate test changes are in a separate commit.  I'll move the sample dependency cleanup into its own commit as well.
